### PR TITLE
EDD-64: Adds client side logging, buttons for opening the log folder, and resetting the application. Adds a downloads report script

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

1. Adding more client side logs for future debugging.
2. Adding a button to the settings dialog to open the log folder, to help a user find that file to send to us
3. Adding a button to the settings dialog to reset the application, or delete the local database file
4. I added a new npm script to generate a report on how many downloads the application has from the GitHub API
5. Added the current application version to the top of the Settings dialog

### What areas of the application does this impact?

Adds logging, adds features to Settings

# Testing

Use the app as normal (start a download, etc.). Open the Settings dialog, and click on Open Log Folder. Inside that folder you will see `main.log`. Open that log to see new log messages

Click on Reset Application to open the Reset Application dialog. Enter `reset application` into the text field to enable the reset button, click on `Reset Application`. This will delete the sqlite3 database file and relaunch the application. Anything saved in the database will reset (window location, download history, token, etc).

Run `npm run download-report` to view application downloads from GitHub

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have bumped the `version` field in package.json and ran `npm install`

### CODE REVIEW

File: src/main/eventHandlers/openUrl.ts [hunk 1]
```diff
@@ -43,6 +43,17 @@ const openUrl = async ({
     const getLinksUrl = searchParams.get('getLinks')
     const clientId = searchParams.get('clientId')
 
+    const paramsToLog = {
```
Comment (macrouch) [inline src/main/eventHandlers/openUrl.ts:46]: Add comment about why I'm logging as an object

File: src/main/eventHandlers/resetApplication.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// @ts-nocheck
+
+import { app, shell } from 'electron'
+import path from 'path'
+
+/**
+ * Opens the download folder associated with the given download ID.
+ * @param {Object} params The params object.
+ * @param {Object} params.database `EddDatabase` instance
+ * @param {String} params.info.downloadId The download ID.
```
Comment (macrouch) [inline src/main/eventHandlers/resetApplication.ts:10]: Remove

File: src/main/eventHandlers/resetApplication.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// @ts-nocheck
+
+import { app, shell } from 'electron'
+import path from 'path'
+
+/**
+ * Opens the download folder associated with the given download ID.
```
Comment (macrouch) [inline src/main/eventHandlers/resetApplication.ts:7]: Update

File: src/app/dialogs/Settings/Settings.jsx [hunk 4]
```diff
@@ -182,6 +195,12 @@ const Settings = ({
 
   return (
     <div>
+      <div className={styles.appVersion}>
```
Comment (eudoroolivares2016) [inline src/app/dialogs/Settings/Settings.jsx:198]: My preference is for this to be on the bottom right of the Settings modal but, I'll defer to more experienced UX developers

File: src/app/dialogs/ResetApplication/ResetApplication.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  const onClickReset = () => {
+    resetApplication()
+  }
+
+  return (
+    <div className={styles.content}>
+      <p>
+        Reseting the application will delete your download history, but any
```
Comment (eudoroolivares2016) [inline src/app/dialogs/ResetApplication/ResetApplication.jsx:44]:
```suggestion
        Resetting the application will delete your download history, but any
```

File: src/main/utils/fetchLinks.ts [hunk 6]
```diff
@@ -38,11 +38,15 @@ const fetchLinks = async ({
 
   // If the getLinks URL is not trusted, don't fetch the links
   if (!isTrustedLink(getLinksUrl)) {
+    const message = `The host [${getLinksUrl}] is not a trusted source and Earthdata Downloader will not continue.\nIf you wish to have this link included in the list of trusted sources please contact us at ${packageDetails.author.email} or submit a Pull Request at ${packageDetails.homepage}.`
+
+    console.log(message)
```
Comment (eudoroolivares2016) [inline src/main/utils/fetchLinks.ts:43]: Remove

File: src/main/utils/database/EddDatabase.ts [hunk 7]
```diff
@@ -16,6 +16,10 @@ class EddDatabase {
     this.tokenId = 1
   }
```
Comment (eudoroolivares2016) [inline src/main/utils/database/EddDatabase.ts:18]:
```suggestion
 /**
   * Destroys the database.
   */
   async destroy() {
    await this.db.destroy()
  }
```

File: scripts/download_report.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+    console.log('Total Downloads: ', totalDownloadCount)
+    console.log('Total Auto Updates: ', totalUpdateCount)
+    console.log('Linux Downloads: ', linuxDownloadCount)
+    console.log('Linux Auto Updates: ', linuxUpdateCount)
+    console.log('MacOS Downloads: ', macDownloadCount)
+    console.log('MacOS Auto Update: ', macUpdateCount)
+    console.log('Windows Downloads: ', windowsDownloadCount)
+    console.log('Windows Auto Update: ', windowsUpdateCount)
+    console.log('downloadCounts: ', JSON.stringify(downloadCounts, null, 2))
```
Comment (trevorlang) [inline scripts/download_report.js:152]: Maybe label this `Download Counts:`? This is the only on not title-cased.

File: src/app/dialogs/Settings/Settings.jsx (hunk 4 above)
Comment (trevorlang) [inline src/app/dialogs/Settings/Settings.jsx:198]: I think this is okay for now but I don't love the design. It feels like a bit of an afterthought. I'll file an improvement ticket so we can clean it up a little later.

File: scripts/download_report.js (hunk 8 above)
Comment (macrouch) [reply scripts/download_report.js:152]: :doh: definitely missed this one

File: src/main/utils/fetchLinks.ts (hunk 6 above)
Comment (macrouch) [reply src/main/utils/fetchLinks.ts:43]: This log is on purpose, to log the error to the log file

File: src/app/dialogs/Settings/Settings.jsx (hunk 4 above)
Comment (macrouch) [reply src/app/dialogs/Settings/Settings.jsx:198]: @trevorlang is going to write a follow up ticket to give it more thought when he has more time

File: src/main/utils/fetchLinks.ts (hunk 6 above)
Comment (eudoroolivares2016) [reply src/main/utils/fetchLinks.ts:43]: Gotcha sorry, I had assumed it was a dev artifact that makes sense here

**Pull request merged**